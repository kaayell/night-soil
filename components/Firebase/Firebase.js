import * as firebase from 'firebase'
import { ANDROID_API_KEY, config, IOS_API_KEY } from './secrets'

export default class Firebase {
  static init () {
    firebase.initializeApp(config)
  }

  static getAuth() {
    return firebase.auth()
  }

  static getUserDetailsRef() {
    let uid = firebase.auth().currentUser.uid
    return firebase.database().ref(`/user/details/${uid}`)
  }

  static savePoop(poopDetails) {
    let currentUser = firebase.auth().currentUser
    firebase.database().ref(`/poop/${currentUser.uid}`).push().set(poopDetails)
  }

  static getPoopsRef(){
    let uid = firebase.auth().currentUser.uid
    return firebase.database().ref(`/poop/${uid}`)
  }

  static async loginWithGoogle () {
    const response = await Expo.Google.logInAsync({
      androidClientId: ANDROID_API_KEY,
      iosClientId: IOS_API_KEY,
      scopes: ['profile', 'email'],
    })

    if (response.type === 'success') {
      const credential = firebase.auth.GoogleAuthProvider.credential(response.idToken, response.accessToken)
      firebase.auth().signInWithCredential(credential)
    }
  }
}