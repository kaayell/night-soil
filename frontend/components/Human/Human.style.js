import { StyleSheet } from 'react-native'
import { POPPINS } from '../StyleGuide/fonts'

export default formStyle = StyleSheet.create({
  text: {
    color: 'white',
    fontFamily: POPPINS
  }
})