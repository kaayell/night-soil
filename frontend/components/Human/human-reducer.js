import * as humanTypes from "./human-types"

export default function humanInfo(state = {}, action = {}) {
    if(action.type === humanTypes.SET_HUMAN_INFO) return action.humanInfo
    return state
}