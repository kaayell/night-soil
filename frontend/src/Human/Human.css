.main_container {
    padding-left: 30px;
    padding-right: 30px;
}

.human_form {
    width: 100%;
    display: flex;
    flex-direction: column;
}


