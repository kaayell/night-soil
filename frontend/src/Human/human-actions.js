import * as humanTypes from "./human-types"

export function setHumanInfo(humanInfo) {
    return {
        type: humanTypes.SET_HUMAN_INFO,
        humanInfo
    }
}