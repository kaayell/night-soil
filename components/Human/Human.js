import React, { Component } from 'react'
import { View } from 'react-native'
import { FormInput, FormLabel, Icon } from 'react-native-elements'
import { POPPINS, POPPINS_MEDIUM } from '../StyleGuide/fonts'
import { BLUE, OFF_WHITE } from '../StyleGuide/colors'
import style from '../StyleGuide/styles'
import Firebase from '../Firebase/Firebase'

export class Human extends Component {
  static navigationOptions = ({navigation}) => ({
    title: 'Profile',
    headerStyle: {
      backgroundColor: BLUE,
      borderBottomWidth: 0
    },
    headerTitleStyle: {
      color: OFF_WHITE,
      fontFamily: POPPINS_MEDIUM
    },
    headerLeft: null,
    headerRight: <Icon name={'clear'} iconStyle={style.icon} color={'white'}
                       onPress={() => navigation.goBack()} underlayColor={BLUE}/>
  })

  componentWillMount () {
    let user = Firebase.getAuth().currentUser
    this.setState({user})
  }

  render () {
    let {user} = this.state
    user = user || {}
    return (
      <View style={{flex: 1, backgroundColor: BLUE, height: '100%'}}>
        <FormLabel labelStyle={{color: 'white', fontSize: 16}} fontFamily={POPPINS}>Name</FormLabel>
        <FormInput disabled value={`${user.displayName}`}/>
        <FormLabel labelStyle={{color: 'white', fontSize: 16}} fontFamily={POPPINS}>Email</FormLabel>
        <FormInput disabled value={user.email}/>
        <FormLabel labelStyle={{color: 'white', fontSize: 16}} fontFamily={POPPINS}>Hourly Rate</FormLabel>
      </View>
    )
  }
}

export default Human