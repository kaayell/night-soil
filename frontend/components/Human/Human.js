import React, { Component } from 'react'
import { connect } from 'react-redux'
import * as api from '../../api/apiClient'
import { setHumanInfo } from './human-actions'
import { View } from 'react-native'
import { FormInput, FormLabel, Icon } from 'react-native-elements'
import { POPPINS, POPPINS_MEDIUM } from '../StyleGuide/fonts'
import { BLUE, OFF_WHITE } from '../StyleGuide/colors'
import style from '../StyleGuide/icon-styles'
import formStyle from './Human.style'

export class Human extends Component {
  static navigationOptions = ({navigation}) => ({
    title: 'Profile',
    headerStyle: {
      backgroundColor: BLUE,
      borderBottomWidth: 0
    },
    headerTitleStyle: {
      color: OFF_WHITE,
      fontFamily: POPPINS_MEDIUM
    },
    headerLeft: null,
    headerRight: <Icon name={'clear'} iconStyle={style.icon} color={'white'}
                       onPress={() => navigation.goBack()} underlayColor={BLUE}/>
  })

  constructor (props) {
    super(props)

    this.state = {
      hourlyRate: ''
    }
  }

  componentWillReceiveProps (nextProps) {
    this.setState({hourlyRate: nextProps.humanInfo.hourlyRate})
  }

  componentWillUnmount () {
    if (this.props.humanInfo.hourlyRate !== this.state.hourlyRate) {
      let humanInfo = {...this.props.humanInfo, ...this.state}
      api.updateHuman(humanInfo)
      this.props.setHumanInfo(humanInfo)
    }
  }

  onValueChange (field, event) {
    const obj = {}
    obj[field] = event
    this.setState(obj)
  }

  render () {
    return (
      <View style={{flex: 1, backgroundColor: BLUE, height: '100%'}}>
        <FormLabel labelStyle={{color: "white", fontSize: 16}} fontFamily={POPPINS}>Name</FormLabel>
        <FormInput disabled value={`${this.props.humanInfo.firstName} ${this.props.humanInfo.lastName}`}/>
        <FormLabel labelStyle={{color: "white", fontSize: 16}} fontFamily={POPPINS}>Email</FormLabel>
        <FormInput disabled value={this.props.humanInfo.email}/>
        <FormLabel labelStyle={{color: "white", fontSize: 16}} fontFamily={POPPINS}>Hourly Rate</FormLabel>
      </View>
    )
  }
}

const mapStateToProps = state => {
  return {
    humanInfo: state.humanInfo
  }
}

export default connect(mapStateToProps, {setHumanInfo})(Human)