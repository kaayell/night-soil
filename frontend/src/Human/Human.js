import React, {Component} from 'react'
import {connect} from 'react-redux'
import * as api from "../api/apiClient"
import {setHumanInfo} from "./human-actions";
import {Content, Form, Input, Item, Label} from "native-base";

export class Human extends Component {
    constructor(props) {
        super(props)

        this.state = {
            hourlyRate: ""
        }
    }

    componentWillReceiveProps(nextProps) {
        this.setState({hourlyRate: nextProps.humanInfo.hourlyRate});
    }

    componentWillUnmount() {
        if (this.props.humanInfo.hourlyRate !== this.state.hourlyRate) {
            let humanInfo = {...this.props.humanInfo, ...this.state}
            api.updateHuman(humanInfo)
            this.props.setHumanInfo(humanInfo)
        }
    }

    onValueChange(field, event) {
        const obj = {};
        obj[field] = event.target.value
        this.setState(obj)
    }

    render() {
        return (
            <Content padder style={{marginTop: 20, backgroundColor: "rgb(248,248,248)"}}>
                <Form>
                    <Item floatingLabel disabled>
                        <Label>Name</Label>
                        <Input disabled
                               placeholder={`${this.props.humanInfo.firstName} ${this.props.humanInfo.lastName}`}/>
                    </Item>
                    <Item floatingLabel disabled>
                        <Label>Email</Label>
                        <Input disabled placeholder={this.props.humanInfo.email}/>
                    </Item>
                    <Item floatingLabel>
                        <Label>Hourly Rate</Label>
                        <Input placeholder={this.props.humanInfo.hourlyRate}/>
                    </Item>
                </Form>
            </Content>
        )
    }

    // render() {
    //     return (
    //         <div className="main_container">
    //             <div className="human_form">
    //                 <TextField floatingLabelText="Name"
    //                            defaultValue={`${this.props.humanInfo.firstName} ${this.props.humanInfo.lastName}`}
    //                            disabled={true}/>
    //                 <TextField floatingLabelText="Email" onChange={this.onValueChange.bind(this, "email")}
    //                            defaultValue={this.props.humanInfo.email}
    //                            disabled={true}/>
    //                 <TextField floatingLabelText="Hourly Rate"
    //                            onChange={this.onValueChange.bind(this, "hourlyRate")}
    //                            defaultValue={this.props.humanInfo.hourlyRate}/>
    //             </div>
    //         </div>
    //     )
    // }
}


const mapStateToProps = state => {
    return {
        humanInfo: state.humanInfo
    }
}

export default connect(mapStateToProps, {setHumanInfo})(Human);