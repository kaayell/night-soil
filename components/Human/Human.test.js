import React from 'react'
import { shallow } from 'enzyme'
import {FormInput, FormLabel} from 'react-native-elements'
import Human from './Human'
import Firebase from '../Firebase/Firebase'

describe('Human', () => {
  let wrapper, mockSet

  beforeEach(() => {
    Firebase.getAuth = jest.fn().mockReturnValue({currentUser: {hi: "hi"}})
    mockSet = jest.fn()
    Firebase.getUserDetailsRef = () => {
      return {
        once: (val) => {
          return {
            then: () => {

            }
          }
        },
        set: (params) => {mockSet(params)}
      }
    }

    wrapper = shallow(<Human/>)
  })

  it('should default render', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('renders with user data', () => {
    wrapper.setState({user: {displayName: 'Mary Hary', email: 'hi@sup.com'}, salary: 10.0})
    expect(wrapper.find(FormLabel).at(0).props().children).toEqual('Mary Hary')
    expect(wrapper.find(FormLabel).at(1).props().children).toEqual('hi@sup.com')
    expect(wrapper.find(FormInput).props().defaultValue).toEqual('10.00')
  })

  describe('on component did mount', () => {
    it('gets user information from firebase and sets it on state', () => {
      wrapper.instance().componentWillMount()
      expect(Firebase.getAuth).toHaveBeenCalled()
      expect(wrapper.state()).toEqual({user: {hi: "hi"}})
    })
  })

  describe('on updating salary', () => {
    it('should set new salary in firebase', () => {
      wrapper.find(FormInput).props().onChangeText("10")
      expect(mockSet).toHaveBeenCalledWith({salary: "10"})
    });
  });
})