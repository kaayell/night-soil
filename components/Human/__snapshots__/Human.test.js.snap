// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Human should default render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#9AC0CD",
      "flex": 1,
      "height": "100%",
    }
  }
>
  <FormLabel
    fontFamily="poppins-regular"
    labelStyle={
      Object {
        "color": "#FAFAFA",
        "fontFamily": "poppins-regular",
        "fontSize": 20,
      }
    }
  />
  <FormLabel
    fontFamily="poppins-regular"
    labelStyle={
      Object {
        "color": "#FAFAFA",
        "fontFamily": "poppins-regular",
        "fontSize": 16,
        "paddingBottom": 30,
      }
    }
  />
  <FormLabel
    fontFamily="poppins-regular"
    labelStyle={
      Object {
        "color": "#FAFAFA",
        "fontSize": 16,
      }
    }
  >
    HOURLY RATE
  </FormLabel>
  <FormInput
    defaultValue="0.00"
    inputStyle={
      Object {
        "color": "#FAFAFA",
        "fontSize": 40,
        "textAlign": "center",
        "width": "50%",
      }
    }
    keyboardType="numeric"
    labelStyle={
      Object {
        "color": "#FAFAFA",
        "fontSize": 16,
      }
    }
    normalizeFontSize={true}
    onChangeText={[Function]}
  />
  <Button
    buttonStyle={
      Object {
        "backgroundColor": "#9AC0CD",
        "borderColor": "#FAFAFA",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginTop": 20,
        "width": "100%",
      }
    }
    onPress={[Function]}
    textStyle={
      Object {
        "color": "#FAFAFA",
        "fontFamily": "poppins-medium",
        "fontSize": 15,
        "fontWeight": "700",
        "textAlign": "center",
      }
    }
    title="SIGN OUT"
  />
</View>
`;
